fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -689,6 +689,11 @@ default Object getComparand() {
         @Nullable
         Object getComparand(@Nullable FDBRecordStoreBase<?> store, @Nullable EvaluationContext context);
 
+        @Nonnull
+        default Comparisons.Comparison withComparand(@Nonnull final Object comparand) {
```

Comment:
The method looks general enough to me that it could have other uses. I think we use this pattern often in Cascades and Heuristics planner.
The method is implemented not only by `ValueComparison`, but also by `SimpleComparison`. It is currently used to compile-time eval a `RangeConstraint` tree structure.
I think it is possible to have a comparison like `SimpleComparison(ComparisonType.LESS_THAN, ConstantObjectValue(const, 0))` and compile-time evaluate it to `SimpleComparison, ComparisonType.LESS_THAN, 42)` given a `ConstantEvaluation` of `const -> [42]`.

Reply:
removed.